src/lib/Component/Content/ContentEditMetaFieldsComponent.php

```diff
@@ -39,17 +40,30 @@ public function render(array $parameters = []): string
         $contentType = $parameters['content_type'];
 
         $metaFieldTypeIdentifiers = $this->getMetaFieldTypeIdentifiers();
+        $metaFieldGroups = $this->configResolver->getParameter(
+            'admin_ui_forms.content_edit.meta_fieldgroup_list'
+        );
+        $metaFieldDefinitionCollection = $contentType->fieldDefinitions->filter(
+            static fn (FieldDefinition $field): bool => true === in_array($field->fieldGroup, $metaFieldGroups),
+        );
```

Comment:
I think you can extract this snippet to new method. It'll be more clear what is it doing.